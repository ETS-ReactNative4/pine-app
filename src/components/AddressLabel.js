import React, { Component } from 'react';
import { View, Text, Clipboard } from 'react-native';
import PropTypes from 'prop-types';
import ToolTip from 'react-native-tooltip';

const STUB_LENGTH = 8;

export default class AddressLabel extends Component {
  constructor() {
    super(...arguments);
    this._onCopy = this._onCopy.bind(this);
  }

  _onCopy() {
    const { address } = this.props;
    Clipboard.setString(address);
  }

  _getShortenedAddress() {
    const { address } = this.props;

    if (!address) {
      return '';
    }

    const start = address.slice(0, STUB_LENGTH);
    const end = address.slice(-STUB_LENGTH);

    return `${start}...${end}`;
  }

  render() {
    const shortenedAddress = this._getShortenedAddress();

    return (
      <View style={this.props.style}>
        <ToolTip
          actions={[
            { text: 'Copy', onPress: this._onCopy }
          ]}
          underlayColor='white'
          activeOpacity={1}
        >
          <Text style={this.props.textStyle}>
            {shortenedAddress}
          </Text>
        </ToolTip>
      </View>
    );
  }
}

AddressLabel.propTypes = {
  style: PropTypes.any,
  textStyle: PropTypes.any,
  address: PropTypes.string
};
