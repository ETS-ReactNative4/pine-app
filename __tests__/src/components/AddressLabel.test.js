import React from 'react';
import renderer from 'react-test-renderer';
import AddressLabel from '../../../src/components/AddressLabel';

describe('AddressLabel', () => {
  it('renders an address correctly', () => {
    const tree = renderer.create(
      <AddressLabel address='moYihV4R1FoSbnukt9sV3WqeFNHnvxfesx' />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders an empty address correctly', () => {
    const tree = renderer.create(
      <AddressLabel />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
