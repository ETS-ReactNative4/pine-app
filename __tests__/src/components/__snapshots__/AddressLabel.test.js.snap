// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressLabel renders an address correctly 1`] = `
<View>
  <RCTToolTipText
    onBlur={[Function]}
    onChange={[Function]}
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        style={null}
      >
        <Text>
          moYihV4R...Hnvxfesx
        </Text>
      </View>
    </View>
  </RCTToolTipText>
</View>
`;

exports[`AddressLabel renders an empty address correctly 1`] = `
<View>
  <RCTToolTipText
    onBlur={[Function]}
    onChange={[Function]}
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        style={null}
      >
        <Text>
          
        </Text>
      </View>
    </View>
  </RCTToolTipText>
</View>
`;
